backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -153,8 +156,22 @@ protected void executeCommand() {
                 ImageStatus.OK,
                 getCompensationContext());
 
+        // Measure volume for new format
+        ActionReturnValue actionReturnValue =
+                runInternalAction(ActionType.MeasureVolume, createMeasureVolumeParameters(),
+                        ExecutionHandler.createDefaultContextForTasks(getContext()));
+
+        if (!actionReturnValue.getSucceeded()) {
+            log.error("Failed to measure volume '{}'", getDiskImage().getImageId());
+            setCommandStatus(CommandStatus.FAILED);
+            return;
+        }
+
+        long requiredSize =  actionReturnValue.getActionReturnValue();
+        requiredSize = Math.max(requiredSize, getDiskImage().getSize());
```

Comment:
what is the scenario in which we would take getDiskImage().getSize() rather than the required size being measured? when exporting to ova we measure the volume to be exported as a qcow2 and don't rely on the size of the original volume (or its chain) - and it seems to work, so either we may have a bug there or we should always rely on the measured result

Reply:
I think this is specific to this case, when converting:
raw -> qcow, and then
qcow -> qcow
The required size can be much smaller, depending on the amount of data which can shrink the size (and fail the conversion because the target is smaller), so I only increase the size if needed

Reply:
As discussed, I confused size with initial size, should be good now